components/system-broker/internal/osb/bind_create_test.go

```diff
@@ -0,0 +1,179 @@
+	})
+
+	t.Run("When async is not supported by platform", func(t *testing.T) {
+		setup()
+		_, err := be.Bind(context.TODO(), instanceID, bindingID, details, false)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "This service plan requires client support for asynchronous service operations")
+	})
+
+	t.Run("When credentials getter returns an error", func(t *testing.T) {
+		setup()
+		fakeCredentialsGetter.FetchPackageInstanceAuthReturns(
+			nil,
+			errors.New("some error"),
+		)
+		_, err := be.Bind(context.TODO(), instanceID, bindingID, details, true)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "while getting package instance credentials from director")
+	})
+
+	t.Run("When raw parameters are not JSON", func(t *testing.T) {
+		setup()
+		details.RawParameters = []byte(`not a json`)
+		fakeCredentialsGetter.FetchPackageInstanceAuthReturns(
+			nil,
+			&notFoundErr{},
+		)
+		packageInstanceAuth.InstanceAuth.Status.Condition = graphql.PackageInstanceAuthStatusConditionPending
+		fakeCredentialsCreator.RequestPackageInstanceCredentialsCreationReturns(
+			packageInstanceAuth,
+			nil,
+		)
+		_, err := be.Bind(context.TODO(), instanceID, bindingID, details, true)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "while unmarshaling raw parameters")
+	})
+
+	t.Run("When OSB context is not JSON", func(t *testing.T) {
+		setup()
+		details.RawParameters = []byte(`{}`)
+		details.RawContext = []byte(`not a json`)
+		fakeCredentialsGetter.FetchPackageInstanceAuthReturns(
+			nil,
+			&notFoundErr{},
+		)
+		packageInstanceAuth.InstanceAuth.Status.Condition = graphql.PackageInstanceAuthStatusConditionPending
+		fakeCredentialsCreator.RequestPackageInstanceCredentialsCreationReturns(
+			packageInstanceAuth,
+			nil,
+		)
+		_, err := be.Bind(context.TODO(), instanceID, bindingID, details, true)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "while unmarshaling raw context")
+	})
+
+	t.Run("When package instance credential requester returns an error", func(t *testing.T) {
+		setup()
+		details.RawContext = []byte(`{"org_guid": "orgID"}`)
+		details.RawParameters = []byte(`{}`)
+		fakeCredentialsGetter.FetchPackageInstanceAuthReturns(
+			nil,
+			&notFoundErr{},
+		)
+		packageInstanceAuth.InstanceAuth.Status.Condition = graphql.PackageInstanceAuthStatusConditionPending
+		fakeCredentialsCreator.RequestPackageInstanceCredentialsCreationReturns(
+			nil,
+			errors.New("some error"),
+		)
+		_, err := be.Bind(context.TODO(), instanceID, bindingID, details, true)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "while requesting package instance credentials creation from director")
+		assert.Equal(t, 1, fakeCredentialsCreator.RequestPackageInstanceCredentialsCreationCallCount())
+		assert.Equal(t, 1, fakeCredentialsGetter.FetchPackageInstanceAuthCallCount())
+	})
+
+	t.Run("When package instance credential requester returns an error", func(t *testing.T) {
```

Comment:
Maybe the name is more appropriate to be: `When package instance auth status is failed`